composables/party/useParty.ts

```diff
@@ -57,23 +60,37 @@ export default <T>({ room, onMessage }: UsePartyParams<T>) => {
     wss.value.get(room.value)?.send(JSON.stringify(value))
   }
 
+  const closeConnection = (room, ws) => {
+    console.log(`[PARTY::CONNECTION] Closing connection with room ${room}`)
+    ws.close()
+    wss.value.delete(room)
+  }
+  const closeAllConnections = () => {
+    for (const [room, ws] of wss.value) {
+      closeConnection(room, ws)
+    }
+  }
+
   watch(
     room,
     (value) => {
-      if (value) {
+      if (value && !disabled) {
         connect(value)
       }
     },
     { immediate: true },
   )
 
-  onBeforeUnmount(() => {
-    for (const [room, ws] of wss.value) {
-      console.log(`[PARTY::CONNECTION] Closing connection with room ${room}`)
-      ws.close()
-      wss.value.delete(room)
-    }
-  })
+  watch(
+    () => preferencesStore.partyMode,
+    (newValue) => {
+      if (newValue === false) {
+        closeAllConnections()
+      }
+    },
+  )
+
```

Comment:
```suggestion
  watch(
    disabled,
    (isDisabled) => {
      if (isDisabled) {
        closeAllConnections()
      }
    },
  )

```

`useParty` needs to be generic, trigger this logic via the param , make it a computed and react here to it 

remove the store